text/0249-v2-experiments.md

```diff
@@ -0,0 +1,137 @@
+---
+feature name: Experimental APIs Post 1.x
+start date: 2020-09-08
+rfc pr: https://github.com/aws/aws-cdk-rfcs/pull/250
+---
+
+# Summary
+
+When CDK v2 is released to General Availability (GA), the main construct library package that we vend will no longer
+contain experimental code. Experimental modules, including modules in Developer Preview, will be vended in a separate
+package. Experimental APIs will be opt-in. Users will not experience breaking changes in minor releases of the CDK
+unless they explicitly opt in by installing an experimental package or by setting a feature flag.
+
+# Motivation
+
+CDK releases contain a combination of stable and experimental features, which has proven to be a pain point for
+customers. The aws-cdk package is released frequently, at least once per week and sometimes more, and each release
+increments the minor version number (e.g 1.59.0 to 1.60.0). In the planned 2.0 release of CDK, the main focus of the
+major version upgrade is to stop packaging modules separately and to include them all in one package called aws-cdk-lib.
+This will solve a number of problems related to peer dependencies that make it harder to vend independent libraries
+based on the CDK, but it does not address the backwards compatibility problem caused by minor releases with breaking
+changes to experimental modules.
+
+The CDK uses an exception to semantic versioning expectations by labeling certain modules as experimental, to allow us
+to make breaking changes to those modules in minor version upgrades. There is precedent for this in other open source
+projects, but for CDK users, it has been a constant source of pain and frustration. Users who do not carefully read and
+understand the documentation simply install packages, copy sample code, make a few tweaks and put the code into
+production. When they later upgrade to a new version, they are surprised to find that their code no longer works. The
+perception of instability has driven some users away, or caused them to limit their usage of CDK to the fundamental L1
+constructs, which do not provide them with the benefits of higher level abstractions.
+
+This RFC proposes that we stop releasing breaking changes in the main packages that we vend that are not hidden by a
+feature flag. A user that installs aws-cdk-lib using npm or pip or any other package manager should be confident that
+there will be no intentional breaking changes in the 2.x line of releases for its lifetime.
+
+# README
```

Comment:
As mentioned in a prior comment, this section is too small and doesn't work properly here.

We should expand this into what an expanded user guide explaining to a new user of the CDK and an existing user of the CDK on the new way to use these packages.

Reply:
I think I will remove this section, since writing new user docs are part of implementation.

Reply:
This is key to describing the change in customer experience. Not sure how you will effectively describe the customer experience without this section.

IIUC, @elad's new working backwards RFC format was written to emphasize on the customer experience.